src/codeflare_sdk/cluster/cluster.py

```diff
@@ -268,19 +268,27 @@ def wait_ready(self, timeout: Optional[int] = None):
         dashboard_ready = False
         status = None
         time = 0
-        while not ready or not dashboard_ready:
+        while not ready:
             status, ready = self.status(print_to_console=False)
-            dashboard_ready = self.is_dashboard_ready()
             if status == CodeFlareClusterStatus.UNKNOWN:
                 print(
                     "WARNING: Current cluster status is unknown, have you run cluster.up yet?"
                 )
-            if not ready or not dashboard_ready:
+            if not ready:
+                if timeout and time >= timeout:
+                    raise TimeoutError(f"wait() timed out after waiting {timeout}s for cluster to be ready")
+                sleep(5)
+                time += 5
+        print("Requested cluster is up and running!")
```

Comment:
Open to thoughts on whether I should create a helper function to avoid repetition. A similar pattern is used when checking for dashboard readiness.